lib/ecto/changeset.ex

```diff
@@ -293,7 +293,7 @@ defmodule Ecto.Changeset do
 
   @type t :: t(Ecto.Schema.t | map | nil)
   @type error :: {String.t, Keyword.t}
-  @type action :: nil | :insert | :update | :delete | :replace | :ignore
+  @type action :: nil | atom
```

Comment:
I wasn't clear about the role that `:ignore` plays in this type, especially given that it was not in `@actions` list.

Reply:
Those are the types used by Ecto, so I would keep them in the list for explicitness and add the catch all, like this:

```suggestion
   @type action :: nil | :insert | :update | :delete | :replace | :ignore | atom
```

Reply:
@josevalim I must admit, I'm still unclear about the role the `:ignore` plays in a changeset, esp given the exception that gets raised in Schema. Under what circumstances would :ignore be used?